FutureNorthants/northants-website
# Backend support for Google Map slice

### Setup
Checkout this branch and run a `./drush.wrapper @docker config:import` and `cache:rebuild`

### Testing
1. Verify that you can now insert a Google Map slice in the body of any page. Ideally a service landing page you've already added a Promotions slice to, with a Hero Image set on the page, as there are a couple of schema tweaks for these elements too.
2. Go to https://core-cms.northants.localhost/graphql and enter the following query:
```graphql
{
  route(path: "your-page-route") {
    node {
      ... on ServiceLandingPageNode {
        heroImage {
          title
          body
          imageLarge
          imageSmall
          imageAltText
        }
      }
      title
      body {
        value
        embeds {
          id
          paragraph {
            ... on PromotionsParagraph {
              promotions {
                id
                imageLarge
                imageMedium
                imageSmall
                imageAltText
                title
                body
                link {
                  title
                  url
                }
              }
            }
            ... on GoogleMapParagraph {
              id
              title
              description
              link {
                title
                url
              }
              iframeHTML
            }
          }
        }
      }
    }
  }
}
```
...and verify it includes the data you expect, including the **body** element of the hero image and promotions area being a string rather than a structure of various fields.
